XCUITests/PrivateBrowsingTest.swift

```diff
@@ -0,0 +1,147 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+let url1 = "www.mozilla.org"
+let url2 = "www.yahoo.com"
+
+let url1Label = "Internet for people, not profit — Mozilla"
+let url2Label = "Yahoo"
+
+class PrivateBrowsingTest: BaseTestCase {
+
+    var navigator: Navigator!
+    var app: XCUIApplication!
+
+    override func setUp() {
+        super.setUp()
+        app = XCUIApplication()
+        navigator = createScreenGraph(app).navigator(self)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+
+    func testPrivateTabDoesNotTrackHistory() {
+        navigator.openURL(urlString: url1)
+        navigator.goto(BrowserTabMenu)
+        // Go to History screen
+        waitforExistence(app.toolbars.buttons["HistoryMenuToolbarItem"])
+        app.toolbars.buttons["HistoryMenuToolbarItem"].tap()
+        navigator.nowAt(NewTabScreen)
+        waitforExistence(app.tables["History List"])
+
+        XCTAssertTrue(app.tables["History List"].staticTexts[url1Label].exists)
+        // History without counting Recently Closed and Synced devices
+        let history = app.tables["History List"].cells.count - 2
+
+        XCTAssertEqual(history, 1, "History entries in regular browsing do not match")
+
+        // Go to Private browsing to open a website and check if it appears on History
+        navigator.goto(PrivateTabTray)
+        navigator.goto(NewPrivateTabScreen)
+        navigator.openURL(urlString: url2, regularBrowsing: false)
+        navigator.goto(BrowserTabMenu)
+        waitforExistence(app.toolbars.buttons["HistoryMenuToolbarItem"])
+        app.toolbars.buttons["HistoryMenuToolbarItem"].tap()
+
+        waitforExistence(app.tables["History List"])
+        XCTAssertTrue(app.tables["History List"].staticTexts[url1Label].exists)
+        XCTAssertFalse(app.tables["History List"].staticTexts[url2Label].exists)
+
+        let privateHistory = app.tables["History List"].cells.count - 2
+        XCTAssertEqual(privateHistory, 1, "History entries in private browsing do not match")
+    }
+
+    func testTabCountShowsOnlyNormalOrPrivateTabCount() {
+        // Open two tabs in normal browsing and check the number of tabs open
+        navigator.openNewURL(urlString: url1)
+        navigator.goto(TabTray)
+        navigator.goto(NewTabScreen)
+        navigator.goto(TabTray)
+
+        waitforExistence(app.collectionViews.cells[url1Label])
+        let numTabs = app.collectionViews.cells.count
+        XCTAssertEqual(numTabs, 2, "The number of regular tabs is not correct")
+
+        // Open one tab in private browsing and check the total number of tabs
+        navigator.goto(NewPrivateTabScreen)
+        navigator.openURL(urlString: url2, regularBrowsing: false)
```

Comment:
Instead of using private: Bool = false, since, as I understand is that you are browsing in private mode, I've changed the parameter name to regularBrowsing, otherwise it's confusing to me if private is set to false, but we are actually browsing in private mode. Did I get it right? does it make sense for you @jhugman? Please let me know otherwise :)